src/main/java/jp/co/yahoo/yosegi/block/PushdownSupportedBlockWriter.java

```diff
@@ -244,47 +245,42 @@ public int size() {
 
     columnTree.create( metaBuffer , dataBuffer );
 
-    byte[] metaBinary = compressor.compress( metaBuffer.getBytes() , 0 , metaBuffer.getLength() );
-
-    byte[] result;
-    if ( dataSize == -1 ) {
-      result = new byte[
-          headerBytes.length
-          + dataBuffer.getLength()
-          + metaBinary.length
-          + Integer.BYTES
-          + Integer.BYTES
-          + ( Integer.BYTES * spreadSizeList.size() ) ];
-    } else {
-      result = new byte[dataSize];
-    }
-
     int offset = 0;
-    System.arraycopy( headerBytes , 0 , result , offset , headerBytes.length );
+    out.write( headerBytes , 0 , headerBytes.length );
     offset += headerBytes.length;
 
-    ByteBuffer wrapBuffer = ByteBuffer.wrap( result );
-    wrapBuffer.putInt( offset , spreadSizeList.size() );
+
+    ByteBuffer blockMetaBuffer = ByteBuffer.allocate(
+        Integer.BYTES + ( Integer.BYTES * spreadSizeList.size() ) + Integer.BYTES );
+    blockMetaBuffer.putInt( spreadSizeList.size() );
+
     offset += Integer.BYTES;
     for ( Integer spreadSize : spreadSizeList ) {
-      wrapBuffer.putInt( offset , spreadSize.intValue() );
+      blockMetaBuffer.putInt( spreadSize.intValue() );
       offset += Integer.BYTES;
     }
 
-    wrapBuffer.putInt( offset , metaBinary.length );
+    byte[] metaBinary = compressor.compress( metaBuffer.getBytes() , 0 , metaBuffer.getLength() );
+    blockMetaBuffer.putInt( metaBinary.length );
     offset += Integer.BYTES;
 
-    System.arraycopy( metaBinary , 0 , result , offset , metaBinary.length );
+    out.write( blockMetaBuffer.array() );
+
+    out.write( metaBinary , 0 , metaBinary.length );
     offset += metaBinary.length;
-    System.arraycopy( dataBuffer.getBytes() , 0 , result , offset , dataBuffer.getLength() );
+    out.write( dataBuffer.getBytes() , 0 , dataBuffer.getLength() );
+    offset += dataBuffer.getLength();
+    if ( dataSize != -1 ) {
+      int nullLength = dataSize - offset;
+      out.write( new byte[nullLength] );
+    } 
 
```

Comment:
Line 273-276 seems pudding. Is my understanding ok?

Reply:
Yes. This line is null filling the block.

Reply:
Thank you. I understand.